ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -186,11 +213,39 @@ object RateLimitingInterceptor {
     )
   }
 
+  private val UnderLimit: Either[String, Unit] = Right(())
+
   val doNonLimit: Set[String] = Set(
     "grpc.reflection.v1alpha.ServerReflection/ServerReflectionInfo",
     "grpc.health.v1.Health/Check",
     "grpc.health.v1.Health/Watch",
   )
+
+  private class OnCloseCallListener[RespT](
```

Comment:
This file is slowly becoming hard to follow. I would consider spliting different unrelated classes into their own files in this module

Reply:
I've created new branch `rate-limit-refactor` that makes things far more manageable. See https://github.com/digital-asset/daml/pull/14311 PR for description.